gtk/src/toga_gtk/app.py

```diff
@@ -236,7 +236,18 @@ def show_cursor(self):
 
     def get_current_window(self):  # pragma: no-cover-if-linux-wayland
         current_window = self.native.get_active_window()._impl
-        return current_window if current_window.interface.visible else None
+        if current_window.interface.visible:
+            # If the focus is on a dialog, then return its host window
+            for window in self.interface.windows:
+                dialog_impl = getattr(window._impl, "dialog_impl", None)
+                if dialog_impl:
+                    if dialog_impl.native.is_visible():  # pragma: no branch
+                        # Marking as no branch, since a window having dialog_impl
+                        # will have its dialog visible, as all dialogs are shown
+                        # in modal style.
+                        return window._impl
+            return current_window
```

Comment:
I'm not clear what's going on with this change - is this something that was revealed as a result of the extra testing? It seems to be trying to protecting against a change of current window while a dialog is visible... but it's not clear to me how that would be possible.